### ISSUE TITLE
LMDB "Transaction ingester aborted" problem in 1.22.0

### ISSUE THREAD

**Issue by tatut** [xtdb/crux#1826]
After running some time, the node seems to stop ingesting.
Await tx causes exception.

The node in question is running on Linux in AWS ECS.
Starting a new instance will restore from checkpoint and indexing works again. 
Don't know the exact pattern that causes the initial problem.

Log has entries with the following exception:

```
java.lang.Exception: Transaction ingester aborted.
at xtdb.node$await_tx.invokeStatic(node.clj:60)
at xtdb.node$await_tx.invoke(node.clj:36)
at xtdb.node.XtdbNode.await_tx(node.clj:158)
at xtdb.node.XtdbNode.await_tx(node.clj:155)
... application code...
Caused by: java.lang.AssertionError: Assert failed: LMDB write lock timeout
(pos? stamp)
at xtdb.lmdb$acquire_write_lock.invokeStatic(lmdb.clj:39)
at xtdb.lmdb$acquire_write_lock.invokePrim(lmdb.clj)
at xtdb.lmdb.LMDBKv.store(lmdb.clj:238)
at xtdb.fork.ForkedKvIndexStoreTx.commit_index_tx(fork.clj:188)
at xtdb.tx.InFlightTx.commit(tx.clj:356)
at xtdb.tx$__GT_tx_ingester$fn__45494$txs_index_fn__45502.invoke(tx.clj:567)
at clojure.lang.AFn.applyToHelper(AFn.java:154)
at clojure.lang.AFn.applyTo(AFn.java:144)
at clojure.core$apply.invokeStatic(core.clj:665)
at clojure.core$apply.invoke(core.clj:660)
at xtdb.tx$__GT_tx_ingester$fn__45494$submit_job_BANG___45495$fn__45496.invoke(tx.clj:544)
at clojure.lang.AFn.call(AFn.java:18)
... 4 common frames omitted
```

The earliest related message seems to be:
```
level | ERROR
logger | xtdb.tx
message | Ingester error occurred
thread | xtdb-tx-processor-1

java.lang.AssertionError: Assert failed: LMDB write lock timeout
(pos? stamp)
at xtdb.lmdb$acquire_write_lock.invokeStatic(lmdb.clj:39)
at xtdb.lmdb$acquire_write_lock.invokePrim(lmdb.clj)
at xtdb.lmdb.LMDBKv.store(lmdb.clj:238)
at xtdb.fork.ForkedKvIndexStoreTx.commit_index_tx(fork.clj:188)
at xtdb.tx.InFlightTx.commit(tx.clj:356)
at xtdb.tx$__GT_tx_ingester$fn__45494$txs_index_fn__45502.invoke(tx.clj:567)
at clojure.lang.AFn.applyToHelper(AFn.java:154)
at clojure.lang.AFn.applyTo(AFn.java:144)
at clojure.core$apply.invokeStatic(core.clj:665)
at clojure.core$apply.invoke(core.clj:660)
at xtdb.tx$__GT_tx_ingester$fn__45494$submit_job_BANG___45495$fn__45496.invoke(tx.clj:544)
at clojure.lang.AFn.call(AFn.java:18)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:833)
```

After that all the await tx calls start failing.

**Comment by jarohen** [comment]
Hey @tatut, thanks for the report. For now (as I'm just about to get on a plane to StrangeLoop!) I've released a snapshot for the LMDB module which reverts most of #1816 - it's version `1.22.0.1-20220921.091757-1`. To configure and use see [here](https://github.com/xtdb/xtdb#pre-release-snapshot-builds).

There was a version bump to LMDB that also went in with 1.22.0 - [LMDB for Java](https://github.com/lmdbjava/lmdbjava) from 0.7.0 to 0.8.2 for M1 support, commit ee70e46060446f9909d53d2898585c4609d30386 - this one's less likely to have caused the regression but we'll keep it in mind.

Thanks again,

James

cc @wotbrew

**Comment by arichiardi** [comment]
Hi there, how can I repro this one? We are also using LMDB and I was wondering if I could help with testing the fix/revert (we'd really love to have M1 in!)

**Comment by wotbrew** [comment]
Just a re-paste of summary I posted in slack: I think the problem is that in 1.21.0 the underlying write lock used by the ingester had no timeout (LMDB controlled the blocking/lock-timeout for regular writes before). The fix #1816 for map resizes also opted all writes in to the 2 minute timeout used for resizes - causing the ingestion timeout if under contention with many or long running queries.

Working on a fix.
